src/components/structures/InteractiveAuth.js

```diff
@@ -61,11 +79,19 @@ export default React.createClass({
             authData: this.props.authData,
             doRequest: this._requestCallback,
             startAuthStage: this._startAuthStage,
+            inputs: this.props.inputs,
+            stateUpdated: this._authStateUpdated,
+            makeRegistrationUrl: this.props.makeRegistrationUrl,
+            matrixClient: this.props.matrixClient,
+            sessionId: this.props.sessionId,
+            clientSecret: this.props.clientSecret,
+            emailSid: this.props.emailSid,
         });
 
         this._authLogic.attemptAuth().then((result) => {
             this.props.onFinished(true, result);
         }).catch((error) => {
+            this.props.onFinished(false, error);
```

Comment:
in an InteractiveAuthDialog, this will close the dialog and swallow the error. Admittedly the current handling of non-interactive-auth errors seems to be totally broken, but this also seems wrong.

(I'm sure I tested non-interactive-auth errors. Perhaps it got broken when matthew added the auth cache stuff?)